src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let stateCopy = { ...state };
+  const stateHistory = [];
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
```

Comment:
Use destruction for `action`
```suggestion
  for (const action of actions) {
    const { type, ........ } = action;
    if (type === 'addProperties') {
```